<launch>
    <!-- Set args -->
    <arg name="use_python" default="true" />
    <arg name="user_name" value="Xiaode" />
    <arg name="marker_frame" value="world" doc="Marker's reference frame"/>
    <arg name="marker_size" default="0.2" />
    <arg name="marker_color_r" default="1" />
    <arg name="marker_color_g" default="0" />
    <arg name="marker_color_b" default="0" />
    <arg name="marker_color_a" default="1" />
    <!-- Start pub_marker node -->
    <node pkg="marker_demo" name="pub_marker" type="pub_marker.py" clear_params="true" output="screen" if="$(arg use_python)">
        <param name="user_name" value="$(arg user_name)" />
        <param name="marker_frame" value="$(arg marker_frame)" />
        <param name="marker_size" type="double" value="$(arg marker_size)" />
        <param name="marker_color_r" type="double" value="$(arg marker_color_r)" />
        <param name="marker_color_g" value="$(arg marker_color_g)" />
        <param name="marker_color_b" value="$(arg marker_color_b)" />
        <param name="marker_color_a" value="$(arg marker_color_a)" />
        <param name="speed" value="1.5" />
    </node>
    <node pkg="marker_demo" name="pub_marker" type="pub_marker" clear_params="true" output="screen" unless="$(arg use_python)">
        <rosparam file="$(find marker_demo)/config/pub_marker_params.yaml" command="load"/>
    </node>
    <!-- Publish TF from world to a_fixed_link -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="world_to_link" args="-0.2 -0.2 0.1 3.1415926 0 0 /world /a_fixed_link" />
    <!-- Start rqt_reconfigure GUI -->
    <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" output="screen"/>
    <!-- Start RViz -->
    <include file="$(find marker_demo)/launch/view_rviz.launch"/>
</launch>
